### PR TITLE
Reduce the Arkansas taxable long term capital gains by the short term capital losses

### LINKED ISSUE #4413: Fix Arkansas capital gains calculation
The 0.741.1 version does not seem to calculate the `ar_taxable_capital_gains` variable correctly.
There need to be more unit tests to test handling of various kinds of losses.

Here is a simple unit test that illustrates the problem:

```
- name: AR taxable capital gains unit test 1
  absolute_error_margin: 0.01
  period: 2021
  input:
    short_term_capital_gains: 800
    state_fips: 5  # AR
  output:  # AR 1000D form says all short-term capital gains are taxable
    ar_taxable_capital_gains: 800
```

PEUS fails this test calculating the variable's value as $400.

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #4413

### CODE REVIEW

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py [hunk 1]
```diff
@@ -10,17 +10,29 @@ class ar_taxable_capital_gains(Variable):
...
 
     def formula(person, period, parameters):
+        # Line 1-3 - long term capital gain or loss
         lt_capital_gains = person("long_term_capital_gains", period)
-        net_st_capital_gains = person("short_term_capital_gains", period)
-        # Can be capital loss
-        net_cap_gain = lt_capital_gains - net_st_capital_gains
+        # Line 4-6 - short term capital loss
+        st_capital_gains = person("short_term_capital_gains", period)
+        capital_loss = -st_capital_gains
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:17]: then you can subtract directly. i think it's sufficiently conventional to consider losses only non-negative values that we can use this directly.
```suggestion
        st_capital_loss = max_(-st_capital_gains, 0)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py [hunk 2]
```diff
@@ -14,12 +14,10 @@ def formula(person, period, parameters):
         lt_capital_gains = person("long_term_capital_gains", period)
         # Line 4-6 - short term capital loss
         st_capital_gains = person("short_term_capital_gains", period)
-        capital_loss = -st_capital_gains
-        has_capital_loss = capital_loss > 0
+        st_capital_loss = max_(-st_capital_gains, 0)
+        has_capital_loss = st_capital_gains < 0
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:18]: Let's distinguish st vs lt throughout
```suggestion
        has_st_capital_loss = st_capital_loss > 0
```

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py [hunk 3]
```diff
@@ -10,24 +10,36 @@ class ar_taxable_capital_gains(Variable):
...
         ).gov.states.ar.tax.income.gross_income.capital_gains
-        capped_net_cap_gain = min_(net_cap_gain, p.exempt.cap)
+        capped_net_cap_gain = min_(net_capital_gain, p.exempt.cap)
+        # Line 8 - Tax rate applied to capital gain
         taxable_capped_net_cap_gain = capped_net_cap_gain * (1 - p.exempt.rate)
-        st_capital_gain = max_(0, net_st_capital_gains)
-        taxable_capital_gain = st_capital_gain + taxable_capped_net_cap_gain
 
-        taxable_capital_loss = -taxable_capital_gain
+        taxable_lg_and_st_capital_gain = where(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:29]:
```suggestion
        taxable_lt_and_st_capital_gain = where(
```

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py [hunk 4]
```diff
@@ -10,24 +10,36 @@ class ar_taxable_capital_gains(Variable):
...
     def formula(person, period, parameters):
+        # Line 1-3 - long term capital gain or loss
         lt_capital_gains = person("long_term_capital_gains", period)
-        net_st_capital_gains = person("short_term_capital_gains", period)
-        # Can be capital loss
-        net_cap_gain = lt_capital_gains - net_st_capital_gains
+        # Line 4-6 - short term capital loss
+        st_capital_gains = person("short_term_capital_gains", period)
+        st_capital_loss = max_(-st_capital_gains, 0)
+        has_st_capital_loss = st_capital_gains < 0
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:18]: move this to the part of the code where it's used

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py [hunk 5]
```diff
@@ -10,24 +10,36 @@ class ar_taxable_capital_gains(Variable):
     defined_for = StateCode.AR
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:10]: add reference

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py [hunk 6]
```diff
@@ -25,14 +28,14 @@ def formula(person, period, parameters):
         capped_net_cap_gain = min_(net_capital_gain, p.exempt.cap)
         # Line 8 - Tax rate applied to capital gain
         taxable_capped_net_cap_gain = capped_net_cap_gain * (1 - p.exempt.rate)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:30]: LMK if this addresses it. Also note that this changes how capital losses are dealt with--they should be negative, but with a floor of the negative cap (an earlier version may have done this).
```suggestion
        # 50% exempt if a gain, otherwise entire loss.
        exempt_fraction = where(capped_net_cap_gain > 0, p.exempt.rate, 0)
        taxable_amount = capped_net_cap_gain * (1 - exempt_fraction)
        # Lines 9-11: Arkansas short term capital gain if any.
        stcg_if_any = max_(st_capital_gains, 0)
        # Line 12: Total taxable gain or loss. Loss is capped.
        total_taxable_cap_gain_or_loss = taxable_amount + stcg_if_any
        loss_cap = p.loss_cap[filing_status]
        return max_(-loss_cap, total_taxable_cap_gain_or_loss)
```

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:30]: Do we want the taxable capital losses to return as negative values? - this will cause issues as we add them to the total taxable income if I am not seeing this wrong

Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:30]: yes, capital losses are negative capital gains...

does the form say to add the absolute value to taxable income?

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:30]: image
[Form](https://www.taxformfinder.org/forms/2023/2023-arkansas-form-ar1000d.pdf#page=1)

Reading the above, it seems like capital loss reduces taxable income (up to $1,500) - so I agree with the code suggestion, unless I am misreading it.

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:30]: Filling out the tax form, this is what I am getting when considered a person with no long term capital gains and only short term losses:
image

Did I miss anything when filling this out? I would make slight adjustments to the code if not

Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:30]: thanks - believe the form aligns with my code suggestion

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:30]: Not quite, in the code suggestion we use:
`        exempt_fraction = where(capped_net_cap_gain > 0, p.exempt.rate, 0)
`
Which only provides the 50% reduction rate if we have net capital gains - I will change it so it also applies to the capital loss situation

Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py]: Sorry I misread. Why multiply 7b by 50%? It's not a gain.

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py]: I see, adjusted, my mistake

**Review by MaxGhenis — APPROVED** [review]
